File: backend/StudioManagement.Data/Entities/Worker.cs
namespace StudioManagement.Data.Entities;

public class Worker : ITenantEntity
{
    public int WorkerId { get; set; }
    public int StudioId { get; set; }
    public string FullName { get; set; } = null!;
    public string? MobileNumber { get; set; }
    public string? Email { get; set; }
    public int? WorkerTypeId { get; set; }
    public bool IsActive { get; set; } = true;
    public string? Notes { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public Studio Studio { get; set; } = null!;
    public WorkerType? WorkerType { get; set; }
    public ICollection<EventWorker> EventWorkers { get; set; } = new List<EventWorker>();
    public ICollection<Expense> Expenses { get; set; } = new List<Expense>();
}

File: src/types/worker.ts
export interface Worker {
  workerId: number;
  fullName: string;
  mobileNumber: string | null;
  email: string | null;
  workerTypeId: number | null;
  workerTypeName: string | null;
  isActive: boolean;
  notes: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface CreateWorkerRequest {
  fullName: string;
  mobileNumber?: string;
  email?: string;
  workerTypeId?: number;
  notes?: string;
}

export type UpdateWorkerRequest = CreateWorkerRequest;
